src/youtube.coffee

```diff
@@ -24,11 +27,21 @@ module.exports = (robot) ->
         key: process.env.HUBOT_YOUTUBE_API_KEY
       })
       .get() (err, res, body) ->
-        videos = JSON.parse(body)
+        robot.logger.debug body
+        if err
+          robot.logger.error err
+          return msg.send err
```

Comment:
It'd be better to `robot.emit 'error', err, msg` to use the builtin error handling. That logs the error by default.